src/libpriv/rpmostree-core.c

```diff
@@ -2926,6 +2956,24 @@ checkout_package_into_root (RpmOstreeContext *self,
                             GCancellable *cancellable,
                             GError      **error)
 {
+  /* If called on compose-side, there may be files to remove from packages specified in the treefile. */
+  g_autoptr(GPtrArray) files_remove_regex = NULL;
+  if (self->treefile_rs)
+    {
+      g_auto(GStrv) files_remove_regex_patterns = 
+        ror_treefile_get_files_remove_regex (self->treefile_rs, dnf_package_get_name (pkg));
+      files_remove_regex = g_ptr_array_new_with_free_func (g_free);
```

Comment:
Hmm, I think this should be

```suggestion
      files_remove_regex = g_ptr_array_new_with_free_func ((GDestroyNotify)g_regex_unref);
```

Reply:
Minor: could also use `g_ptr_array_new_full` here so that it pre-allocates the array since we know how many items we'll add.

Reply:
Ah ok. Didn't realize GRegex had additional resources allocated for it.